### ISSUE TITLE
Shape consistency check needed.

### ISSUE THREAD

**Issue by Godricly** [dmlc/gluon-cv#137]
if I don't use any transform. it will crash with batch size greater than 1.
[link](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/data/dataloader.py#L9).
shape consistency check is missing before stacking.
It is able to get the correct shape, but getting the data leads to crash.

```python
    batch_size = 2  # for tutorial, we use smaller batch-size
    num_workers = 0  # you can make it larger(if your CPU has more cores) to accelerate data loading
    train_loader = DetectionDataLoader(train_dataset, batch_size, shuffle=True,
                                       last_batch='rollover', num_workers=num_workers)
    for ib, batch in enumerate(train_loader):
        if ib > 3:
            break
        print batch[0].asnumpy()
```
@zhreshold

**Comment by Godricly** [comment]
The problem is with the nd.stack function.

**Comment by zhreshold** [comment]
We will going to add some default batchify behaviors such as stack, padding, and list grouping without modification. So stay tuned for now.
